week-4/Homework/mandatory/3-water-bottle.js

```diff
@@ -50,7 +65,7 @@ Extra question:
   Leave your answer below:
 */
 
-// HERE COMES YOUR ANSWER
+// HERE COMES YOUR ANSWER : I don't exactly know why, but I find it clearer and more conveinent to use this rather than the object name when i'm referring to things inside it.

```

Comment:
One reason is that by using `this` keeps the code more enclosed/encapsulated, there is no outside dependency. Because if it refers to itself, `this` makes it more resilient, also it is more pragmatic solution. E.g. if you rename the variable name then you should also rename the corresponding variable name inside. However, by using `this` you don't need to.